### DISCUSSION TITLE
Get a surface mesh's inner point

### DISCUSSION THREAD

**Discussion by GenieYoung** [CGAL/cgal#7846] [Q&A]
Is there any method that can get a inner point of a surface mesh? The locate method in Polygon_mesh_processing seems not provide such method.

**Comment by pentacular** [comment]
Please take a look at [Side_of_triangle_mesh.](https://doc.cgal.org/latest/Polygon_mesh_processing/classCGAL_1_1Side__of__triangle__mesh.html)

**Reply by GenieYoung** [reply]
Thanks, i know this method can fast check weather a point is in/out a surface mesh, but how to get a inner point directly?    I think a direct method is to choose a face and pick up points along the normal direction recursively (half split),  and terminate when a point is checked to be  inside this mesh.  I want konw weather cgal provide a more simple method.

**Reply by pentacular** [reply]
I think it would help to understand why you want to find an arbitrary point that is inside the mesh?

**Reply by GenieYoung** [reply]
In some situation, i will construct two isomorphic(same geometry) mesh(complex), but their volumes has different order,  i want to link the corresponding volume, so i use inner point to locate which volume is same.  Inner point can also be used to mark a region in mesh generation method
